include/exadg/structure/time_integration/time_int_gen_alpha.cpp

```diff
@@ -124,7 +124,24 @@ TimeIntGenAlpha<dim, Number>::do_timestep_solve()
   VectorType const_vector, rhs;
   const_vector.reinit(displacement_n);
   rhs.reinit(displacement_n);
-  this->compute_const_vector(rhs, displacement_n, velocity_n, acceleration_n);
+
+  // add const_vector resulting from acceleration and weak damping terms
+  double total_scaling_factor_mass = this->get_scaling_factor_mass_from_acceleration();
+  this->compute_const_vector_acceleration_remainder(rhs /* placeholder */,
+                                                    displacement_n,
+                                                    velocity_n,
+                                                    acceleration_n);
+  if(this->param.weak_damping_coefficient > 1e-20 * this->param.density)
+  {
+    this->compute_const_vector_velocity_remainder(const_vector /* placeholder */,
+                                                  displacement_n,
+                                                  velocity_n,
+                                                  acceleration_n);
+    rhs.add(param.weak_damping_coefficient, const_vector);
+    const_vector = 0; // set to 0 since we apply_mass_operator
+    total_scaling_factor_mass +=
+      param.weak_damping_coefficient * this->get_scaling_factor_mass_from_velocity();
```

Comment:
I would shift these lines upwards, because above you also first compute the scaling factor and then the vector.